samples/bookinfo/platform/kube/bookinfo-add-serviceaccount.yaml

```diff
@@ -3,12 +3,19 @@ kind: ServiceAccount
 metadata:
   name: bookinfo-productpage
 ---
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: productpage-v1
+  labels:
+    app: productpage
+    version: v1
 spec:
   replicas: 1
+  selector:
+    matchLabels:
```

Comment:
What are these matchLabels used for? Do we need them? Do they need to be a 1-1 copy of the pod labels?

Reply:
In k8s documentation:

> In API version `apps/v1`, `.spec.selector` and `.metadata.labels` do not default to `.spec.template.metadata.labels` if not set. So they must be set explicitly. Also note that `.spec.selector` is immutable after creation of the Deployment in `apps/v1`.

[https://kubernetes.io/docs/concepts/workloads/controllers/deployment/](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)